packages/malloy/src/lang/ast/sources/table-source.ts

```diff
@@ -60,3 +60,41 @@ export class TableSource extends Source {
     return ErrorFactory.structDef;
   }
 }
+
+export class TableMethodSource extends TableSource {
+  elementType = 'tableMethodSource';
+  constructor(
+    readonly connectionName: ModelEntryReference,
+    readonly name: string
+  ) {
+    super();
+  }
+
+  getFullName(): string | undefined {
+    const connection = this.modelEntry(this.connectionName);
+    if (connection === undefined) {
+      this.connectionName.log(
+        `${this.connectionName.refString} is not defined`
+      );
+      return undefined;
+    } else if (connection.entry.type !== 'connection') {
+      this.connectionName.log(
+        `${this.connectionName.refString} is not a connection`
+      );
+      return undefined;
+    }
+    return `${this.connectionName.refString}:${this.name}`;
+  }
+}
+
+export class TableFunctionSource extends TableSource {
+  elementType = 'tableFunctionSource';
+  constructor(readonly name: string) {
+    super();
+  }
+
+  getFullName(): string | undefined {
+    // TODO return this parsed...
+    return this.name;
```

Comment:
I would like for this method really to be `getTableInfo(): { connectionName: string; tableName: string }`. This would possibly mean having to a) change the translator's API to make it so the request for schemas can have two pieces of info (both the connection name and the table name), rather than one combined string b) accept the split name and then package it back up into one string for use with the translator...